common/lib/xmodule/xmodule/js/src/video/03_video_player.js

```diff
@@ -381,73 +372,8 @@ function(HTML5Video, Resizer) {
         }
     }
 
-    function setPlaybackRate(newSpeed, useCueVideoById) {
-        var duration = this.videoPlayer.duration(),
-            time = this.videoPlayer.currentTime,
-            methodName, youtubeId;
-
-        // There is a bug which prevents YouTube API to correctly set the speed
-        // to 1.0 from another speed in Firefox when in HTML5 mode. There is a
-        // fix which basically reloads the video at speed 1.0 when this change
-        // is requested (instead of simply requesting a speed change to 1.0).
-        // This has to be done only when the video is being watched in Firefox.
-        // We need to figure out what browser is currently executing this code.
-        //
-        // TODO: Check the status of
-        // http://code.google.com/p/gdata-issues/issues/detail?id=4654
-        // When the YouTube team fixes the API bug, we can remove this
-        // temporary bug fix.
-
-        // If useCueVideoById is true it will reload video again.
-        // Used useCueVideoById to fix the issue video not playing if we change
-        // the speed before playing the video.
```

Comment:
@muzaffaryousaf When the original code was written, the YouTube API had a bug where, in HTML5 mode, you couldn't switch back to speed 1.0x from another speed. The workaround was to just reload the video  (which reset the speed) and cue it to the previous timestamp. However, the bug in the YouTube API has since been fixed, and the workaround became the source of the problem: in Firefox, if you set a video's playback rate to anything other than 1.0x, you could not switch back to 1.0x. My code fixes this bug in light of the updated YouTube API.